framework/tlvf/src/src/WSC/AttrList.cpp

```diff
@@ -0,0 +1,504 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2019 Tomer Eliyahu (Intel)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include <tlvf/WSC/AttrList.h>
+#include <tlvf/tlvflogging.h>
+
+using namespace WSC;
+
+std::shared_ptr<AttrList> AttrList::create(ieee1905_1::tlvWsc &tlv, const config &cfg)
+{
+    std::shared_ptr<AttrList> attributes = nullptr;
+    if (cfg.msg_type == eWscMessageType::WSC_MSG_TYPE_M1) {
+        attributes = std::make_shared<m1>(tlv.payload(), tlv.payload_length(), false);
+        if (!attributes->init(cfg))
+            return nullptr;
+        attributes->finalize();
+        tlv.addInnerClassList(attributes);
+    } else if (cfg.msg_type == eWscMessageType::WSC_MSG_TYPE_M2) {
+        attributes = std::make_shared<m2>(tlv.payload(), tlv.payload_length(), false);
+        if (!attributes->init(cfg))
+            return nullptr;
+        attributes->finalize();
+        tlv.addInnerClassList(attributes);
```

Comment:
This part is duplicated between the M1 and M2. You can move it outside the if statement.